Source/JavaScriptCore/b3/B3LowerInt64.cpp

```diff
@@ -963,9 +963,22 @@ class LowerInt64 {
             return;
         }
         case Load: {
-            if (m_value->type() != Int64)
-                return;
+            // Lower Int64 loads to Int32 loads.
+            //
+            // On ARMv7, loads of Doubles/Floats will fault on unaligned access,
+            // so lower those to Int32 loads as well.
             MemoryValue* memory = m_value->as<MemoryValue>();
+            if (isARM_THUMB2() && m_value->type() == Float) {
```

Comment:
I don't think isARM_THUMB2 check is good. We should have a characteristics and abstract it in CPU.h and use that function instead of isARM_THUMB2

Reply:
Fair enough. Uploaded a new version that defines/uses `hasUnalignedFPMemoryAccess()`.